### PR TITLE
RolePermissionMapping API

### PR DESCRIPTION
**Pull request by sumanvpacewisdom**
Story No - 1048 
Assign Permisson to Role

### CODE REVIEW

File: src/controllers/v1/rolePermissionMapping.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	 * Create rolePermission.
+	 * @method
+	 * @name create
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission creation object.
+	 */
+
+	async create(req) {
+		try {
+			const createRolePermission = await RolePermissionService.create(req.params.id, req.body.permission_id)
```
Comment (aks30) [inline src/controllers/v1/rolePermissionMapping.js:15]: @sumanvpacewisdom it should be camel case - rolePermissionService

File: src/controllers/v1/rolePermissionMapping.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+module.exports = class RolePermission {
+	/**
+	 * Create rolePermission.
+	 * @method
+	 * @name create
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission creation object.
+	 */
+
+	async create(req) {
```
Comment (aks30) [inline src/controllers/v1/rolePermissionMapping.js:13]: @sumanvpacewisdom should not we have update also?

File: src/database/migrations/20231219073900-role_permission_mapping.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+			},
+			actions: {
+				allowNull: false,
+				type: Sequelize.ARRAY(Sequelize.STRING),
+			},
+			created_at: {
+				allowNull: false,
+				type: Sequelize.DATE,
+			},
+			updated_at: {
```
Comment (aks30) [inline src/database/migrations/20231219073900-role_permission_mapping.js:29]: @sumanvpacewisdom - we should add created_by and updated_by as well which should be set to who is creating and updating these data.

File: src/database/seeders/20231211091753-permissions.js [hunk 4]
```diff
@@ -7,8 +7,8 @@ module.exports = {
 			const permissionsData = [
 				{
 					code: 'all_session',
```
Comment (aks30) [inline src/database/seeders/20231211091753-permissions.js:9]: @sumanvpacewisdom code should be manage_session right?

File: src/services/rolePermissionMapping.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	 * @method
+	 * @name create
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission creation object.
+	 */
+
+	static async create(role_id, permission_id) {
+		try {
+			const permission = await permissionsQueries.findPermissionId(permission_id)
+			const Data = {
```
Comment (aks30) [inline src/services/rolePermissionMapping.js:20]: @sumanvpacewisdom this should be small case

File: src/services/rolePermissionMapping.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	 * Create rolePermission.
+	 * @method
+	 * @name create
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission creation object.
+	 */
+
+	static async create(role_id, permission_id) {
+		try {
+			const permission = await permissionsQueries.findPermissionId(permission_id)
```
Comment (aks30) [inline src/services/rolePermissionMapping.js:19]: @sumanvpacewisdom what if permission is not found?

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 3 Jan

**Comment by sumanvpacewisdom** [comment]
@nevil-mathew Please review this PR and let me now if there is any changes.

File: src/database/migrations/20231219073900-role_permission_mapping.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+		await queryInterface.createTable('role_permission_mapping', {
+			role_id: {
+				allowNull: false,
+				type: Sequelize.INTEGER,
+				primaryKey: true,
+			},
+			permission_id: {
+				allowNull: false,
+				type: Sequelize.INTEGER,
+				primaryKey: true,
```
Comment (nevil-mathew) [inline src/database/migrations/20231219073900-role_permission_mapping.js:15]: Table shouldn't have multiple primary keys.

File: src/database/models/rolePermissionMapping.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+		'RolePermission',
+		{
+			role_id: {
+				type: DataTypes.INTEGER,
+				primaryKey: true,
+				allowNull: false,
+			},
+			permission_id: {
+				type: DataTypes.INTEGER,
+				primaryKey: true,
```
Comment (nevil-mathew) [inline src/database/models/rolePermissionMapping.js:12]: Table shouldn't have multiple primary keys.

File: src/database/seeders/20231219074003-role_permission_mapping.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+				},
+			]
+			await queryInterface.bulkInsert('role_permission_mapping', rolePermissionsData)
+		} catch (error) {
+			console.log(error)
+		}
+	},
+
+	down: async (queryInterface, Sequelize) => {
+		await queryInterface.dropTable('role_permission_mapping')
```
Comment (nevil-mathew) [inline src/database/seeders/20231219074003-role_permission_mapping.js:34]: Why are we dropping table in seeder?
You should only remove the inserted data.

File: src/services/rolePermissionMapping.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+		try {
+			const filter = { role_id, permission_id }
+			const rolepermissionmapping = await rolePermissionMappingQueries.delete(filter)
+			return common.successResponse({
+				statusCode: httpStatusCode.created,
+				message: 'ROLE_PERMISSION_MAPPING_DELETED_SUCCESSFULLY',
+				result: {},
+			})
+		} catch (error) {
+			if (error instanceof UniqueConstraintError) {
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:75]: Why is there a UniqueConstraintError in delete api?

File: src/services/rolePermissionMapping.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+	/**
+	 * Delete rolePermission.
+	 * @method
+	 * @name delete
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission deletion object.
+	 */
+
+	static async delete(role_id, permission_id) {
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:65]: use camelCase

File: src/services/rolePermissionMapping.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+module.exports = class modulesHelper {
+	/**
+	 * Create rolePermission.
+	 * @method
+	 * @name create
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission creation object.
+	 */
+
+	static async create(role_id, permission_id, id) {
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:17]: use camelCase

File: src/validators/v1/rolePermissionMapping.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+	create: (req) => {
+		req.checkBody('permission_id')
```
Comment (nevil-mathew) [inline src/validators/v1/rolePermissionMapping.js:3]: Are we only expecting permission_id? no other data is required?

File: src/database/seeders/20231211091753-permissions.js [hunk 14]
```diff
@@ -6,9 +6,9 @@ module.exports = {
 		try {
```
Comment (nevil-mathew) [inline src/database/seeders/20231211091753-permissions.js:6]: Why are we dropping table in seeder?
You should only remove the inserted data.

File: src/controllers/v1/rolePermissionMapping.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+		} catch (error) {
+			return error
+		}
+	}
+
+	/**
+	 * Delete rolePermission.
+	 * @method
+	 * @name delete
+	 * @param {Object} req - Request data.
```
Comment (nevil-mathew) [inline src/controllers/v1/rolePermissionMapping.js:30]: add req.params.id etc as well

File: src/services/rolePermissionMapping.js [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+				})
+			}
+			const data = {
+				role_id,
+				permission_id,
+				module: permission.module,
+				actions: permission.actions,
+				created_by: id,
+			}
+			const rolepermissionmapping = await rolePermissionMappingQueries.create(data)
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:34]: use camelCase

File: src/services/rolePermissionMapping.js [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+	 * @method
+	 * @name delete
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission deletion object.
+	 */
+
+	static async delete(role_id, permission_id) {
+		try {
+			const filter = { role_id, permission_id }
+			const rolepermissionmapping = await rolePermissionMappingQueries.delete(filter)
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:68]: use camelCase

File: src/services/rolePermissionMapping.js [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+				module: permission.module,
+				actions: permission.actions,
+				created_by: id,
+			}
+			const rolepermissionmapping = await rolePermissionMappingQueries.create(data)
+			return common.successResponse({
+				statusCode: httpStatusCode.created,
+				message: 'ROLE_PERMISSION_MAPPING_CREATED_SUCCESSFULLY',
+				result: {
+					role_Id: rolepermissionmapping.role_id,
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:39]: use camel_case

File: src/database/seeders/20231219074003-role_permission_mapping.js [hunk 19]
```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		try {
+			const rolePermissionsData = [
+				{
+					role_id: 6,
+					permission_id: 1,
```
Comment (nevil-mathew) [inline src/database/seeders/20231219074003-role_permission_mapping.js:10]: We cannot expect the role_id etc to be 6 in all cases right?

File: src/services/rolePermissionMapping.js [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+					roleId: rolePermissionMapping.role_id,
+					permissionId: rolePermissionMapping.permission_id,
+					module: rolePermissionMapping.module,
+					actions: rolePermissionMapping.actions,
+				},
+			})
+		} catch (error) {
+			if (error instanceof UniqueConstraintError) {
+				return common.failureResponse({
+					message: 'ROLE_PERMISSION_MAPPING_ALREADY_EXISTS',
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:48]: this message is missing from en.json

File: src/services/rolePermissionMapping.js [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+	 * @returns {JSON} - rolePermission deletion object.
+	 */
+
+	static async delete(roleId, permissionId) {
+		try {
+			const filter = { role_id: roleId, permission_id: permissionId }
+			const rolePermissionMapping = await rolePermissionMappingQueries.delete(filter)
+			if (rolePermissionMapping == 0) {
+				return common.failureResponse({
+					message: 'ROLE_PERMISSION_MAPPING_ALREADY_EXISTS',
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:71]: Please update this message to be something meaningful

File: src/services/rolePermissionMapping.js [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+const { Op } = require('sequelize')
+
+module.exports = class modulesHelper {
+	/**
+	 * Create rolePermission.
+	 * @method
+	 * @name create
+	 * @param {Object} req - Request data.
+	 * @returns {JSON} - RolePermission creation object.
+	 */
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:15]: fix the JSDoc

File: src/validators/v1/rolePermissionMapping.js [hunk 23]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+	create: (req) => {
```
Comment (nevil-mathew) [inline src/validators/v1/rolePermissionMapping.js:2]: validate req.params.id

File: src/validators/v1/rolePermissionMapping.js [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+		req.checkBody('permission_id')
+			.trim()
+			.notEmpty()
+			.withMessage('permission_id field is empty')
+			.matches(/^[0-9]+$/)
+			.withMessage('permission_id is invalid, must not contain spaces')
+	},
+
+	delete: (req) => {
```
Comment (nevil-mathew) [inline src/validators/v1/rolePermissionMapping.js:11]: validate req.params.id

File: src/requests/user.js [hunk 25]
```diff
@@ -6,6 +6,7 @@
  */
 
 // Dependencies
+require('dotenv').config()
```
Comment (nevil-mathew) [inline src/requests/user.js:9]: do we need this here ?

File: src/database/seeders/20231219074003-role_permission_mapping.js [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+			})
+		)
+	} catch (error) {
+		throw error
+	}
+}
+
+const getPermissionId = async (code, actions) => {
+	try {
+		// Customize this query based on your requirements
```
Comment (nevil-mathew) [inline src/database/seeders/20231219074003-role_permission_mapping.js:49]: remove this comment

File: src/constants/common.js [hunk 27]
```diff
@@ -171,6 +173,7 @@ module.exports = {
 	sessionCompleteEndpoint: '/mentoring/v1/sessions/completed/',
 	INACTIVE_STATUS: 'INACTIVE',
 	ACTIVE_STATUS: 'ACTIVE',
+	SEARCH: 'a',
```
Comment (nevil-mathew) [inline src/constants/common.js:176]: What does this do ? If not needed remove this please

File: src/requests/user.js (hunk 25 above)
Comment (sumanvpacewisdom) [reply src/requests/user.js:9]: yes , else it won't take the userservicehost and url @nevil-mathew

File: src/services/rolePermissionMapping.js [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+	 * @returns {JSON} - rolePermission deletion object.
+	 */
+
+	static async delete(roleId, permissionId) {
+		try {
+			const filter = { role_id: roleId, permission_id: permissionId }
+			const rolePermissionMapping = await rolePermissionMappingQueries.delete(filter)
+			if (rolePermissionMapping == 0) {
+				return common.failureResponse({
+					message: 'ROLE_PERMISSION_ALREADY_EXISTS',
```
Comment (nevil-mathew) [inline src/services/rolePermissionMapping.js:74]: This message don't clearly communicate the issue

File: src/constants/common.js (hunk 27 above)
Comment (sumanvpacewisdom) [reply src/constants/common.js:176]: I fetching Search string from common . i need this @nevil-mathew

File: src/services/rolePermissionMapping.js (hunk 28 above)
Comment (sumanvpacewisdom) [reply src/services/rolePermissionMapping.js:74]: U mean to say the message inside en file is not proper or here @nevil-mathew

File: src/test.js [hunk 29]
```diff
@@ -0,0 +1,60 @@
+require('module-alias/register')
```
Comment (nevil-mathew) [inline src/test.js:1]: Why is this file in src?

File: src/services/mentees.js [hunk 30]
```diff
@@ -58,10 +58,33 @@ module.exports = class MenteesHelper {
...
+
+		const uniqueModules = new Set()
+		const uniqueActions = new Set()
+
+		permissionAndModules.forEach((rolePermission) => {
+			uniqueModules.add(rolePermission.dataValues.module)
+			rolePermission.dataValues.actions.forEach((action) => uniqueActions.add(action))
+		})
+
+		const actionAndModules = {
```
Comment (nevil-mathew) [inline src/services/mentees.js:74]: Module and actions should be an array of objects

File: src/services/mentees.js [hunk 31]
```diff
@@ -31,7 +31,7 @@ module.exports = class MenteesHelper {
 	 * @param {String} userId - user id.
 	 * @returns {JSON} - profile details
 	 */
-	static async read(id, orgId) {
+	static async read(roleId, id, orgId) {
```
Comment (nevil-mathew) [inline src/services/mentees.js:34]: We're passing role id here right ?

File: src/services/mentees.js [hunk 32]
```diff
@@ -58,10 +58,33 @@ module.exports = class MenteesHelper {
 
 		const totalSession = await sessionAttendeesQueries.countEnrolledSessions(id)
 
+		const [firstRole] = roleId
+		const userRoles = firstRole?.id
```
Comment (nevil-mathew) [inline src/services/mentees.js:62]: User can have multiple roles, this doesnt work with that, use map to extract all the ids

File: src/middlewares/authenticator.js [hunk 33]
```diff
@@ -122,6 +146,27 @@ module.exports = async function (req, res, next) {
 			decodedToken.data.organization_id = user.data.result.organization_id
 		}
 
+		if (apiPermissions) {
```
Comment (nevil-mathew) [inline src/middlewares/authenticator.js:149]:
```
Store user available module and permissions in req.module[] and req.permissions[] respectively  
```
This logic is missing

**Comment by sumanvpacewisdom** [comment]
@nevil-mathew I have made the changes for the comments , please check and let me know .

File: src/services/mentees.js [hunk 34]
```diff
@@ -58,31 +58,33 @@ module.exports = class MenteesHelper {
 
 		const totalSession = await sessionAttendeesQueries.countEnrolledSessions(id)
 
-		const [firstRole] = roleId
-		const userRoles = firstRole?.id
-
+		const userRoles = roles.map((role) => role.id)
```
Comment (nevil-mathew) [inline src/services/mentees.js:61]: use "userRoleIds" as the variable name.

File: src/services/mentees.js [hunk 35]
```diff
@@ -58,31 +58,33 @@ module.exports = class MenteesHelper {
...
+			} else {
+				permissionsByModule[module] = { module, actions: [...actions] }
+			}
 		})
 
-		const actionAndModules = {
-			module: Array.from(uniqueModules),
-			actions: Array.from(uniqueActions),
-		}
+		const resultArray = Object.entries(permissionsByModule).map(([key, value]) => ({
```
Comment (nevil-mathew) [inline src/services/mentees.js:76]: use permissions as the variable name

File: src/services/mentors.js [hunk 36]
```diff
@@ -578,23 +578,25 @@ module.exports = class MentorsHelper {
 
 			const totalSession = await sessionAttendeesQueries.countEnrolledSessions(id)
 
-			const [firstRole] = roleId
-			const userRoles = firstRole?.id
-
+			const userRoles = roles.map((role) => role.id)
```
Comment (nevil-mathew) [inline src/services/mentors.js:581]: use "userRoleIds" as the variable name.

File: src/services/mentors.js [hunk 37]
```diff
@@ -578,23 +578,25 @@ module.exports = class MentorsHelper {
...
+				} else {
+					permissionsByModule[module] = { module, actions: [...actions] }
+				}
 			})
 
-			const actionAndModules = {
-				module: Array.from(uniqueModules),
-				actions: Array.from(uniqueActions),
-			}
+			const resultArray = Object.entries(permissionsByModule).map(([key, value]) => ({
```
Comment (nevil-mathew) [inline src/services/mentors.js:596]: use "permissions" as the variable name

File: src/database/queries/rolePermissionMapping.js [hunk 38]
```diff
@@ -0,0 +1,30 @@
...
+		return deletedRows
+	} catch (error) {
+		throw error
+	}
+}
+
+exports.find = async (filter) => {
+	try {
+		return RolePermissionMapping.findAll({
+			where: { role_id: filter },
```
Comment (nevil-mathew) [inline src/database/queries/rolePermissionMapping.js:25]: Make this generic

**Comment by nevil-mathew** [comment]
Reviewed and approved

**Comment by sumanvpacewisdom** [comment]
@rakeshSgr Nevil has approved , please take it forward .

File: src/database/migrations/20231211091634-add-permissions-table.js [hunk 39]
```diff
@@ -38,9 +44,10 @@ module.exports = {
 				type: Sequelize.DATE,
 			},
 		})
-		await queryInterface.addIndex('permissions', ['code'], {
-			unique: true,
-			name: 'unique_code',
+		await queryInterface.addIndex('permissions', {
```
Comment (rakeshSgr) [inline src/database/migrations/20231211091634-add-permissions-table.js:47]: @sumanvpacewisdom remove unique for api_path

File: src/database/migrations/20240117125339-add-permissions-data.js [hunk 40]
```diff
@@ -0,0 +1,234 @@
...
+			console.log(error)
+		}
+	},
+
+	down: async (queryInterface, Sequelize) => {
+		await queryInterface.bulkDelete('permissions', null, {})
+	},
+}
+
+//{
```
Comment (rakeshSgr) [inline src/database/migrations/20240117125339-add-permissions-data.js:189]: @sumanvpacewisdom remove un wanted code

Comment (rakeshSgr) [inline src/database/migrations/20240117125339-add-permissions-data.js:189]: @sumanvpacewisdom remove un wanted code

File: src/database/models/permissions.js [hunk 41]
```diff
@@ -31,7 +37,15 @@ module.exports = (sequelize, DataTypes) => {
 			modelName: 'Permission',
 			tableName: 'permissions',
 			freezeTableName: true,
-			indexes: [{ unique: true, fields: ['code'] }],
+			indexes: [
+				{
+					unique: true,
+					fields: ['code', 'api_path'],
```
Comment (rakeshSgr) [inline src/database/models/permissions.js:43]: @sumanvpacewisdom remove unique for api_path

Comment (rakeshSgr) [inline src/database/models/permissions.js:43]: @sumanvpacewisdom remove unique for api_path

File: src/database/seeders/20231219074003-role_permission_mapping.js [hunk 42]
```diff
@@ -0,0 +1,323 @@
...
+		throw error
+	}
+}
+
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		try {
+			await getRoleIds()
+
+			const rolePermissionsData = [
```
Comment (rakeshSgr) [inline src/database/seeders/20231219074003-role_permission_mapping.js:59]: @sumanvpacewisdom do we need this seeder ?

Comment (rakeshSgr) [inline src/database/seeders/20231219074003-role_permission_mapping.js:59]: @sumanvpacewisdom move this to migrations

File: src/services/mentees.js [hunk 43]
```diff
@@ -59,10 +59,36 @@ module.exports = class MenteesHelper {
 
 		const totalSession = await sessionAttendeesQueries.countEnrolledSessions(id)
 
+		const userRoleIds = roles.map((role) => role.id)
+		const filter = { role_id: userRoleIds }
+		const permissionAndModules = await rolePermissionMappingQueries.find(filter)
+		const permissionsByModule = {}
+
+		permissionAndModules.forEach((rolePermission) => {
```
Comment (rakeshSgr) [inline src/services/mentees.js:67]: @sumanvpacewisdom why this changes ?

File: src/services/mentees.js [hunk 44]
```diff
@@ -59,10 +59,36 @@ module.exports = class MenteesHelper {
...
+			if (permissionsByModule[module]) {
+				permissionsByModule[module].actions.push(...actions)
+			} else {
+				permissionsByModule[module] = { module, actions: [...actions] }
+			}
+		})
+
+		const permissions = Object.entries(permissionsByModule).map(([key, value]) => ({
+			module: value.module,
+			actions: value.actions,
```
Comment (rakeshSgr) [inline src/services/mentees.js:80]: @sumanvpacewisdom actions changed to request_type right

File: src/services/mentees.js [hunk 45]
```diff
@@ -59,10 +59,36 @@ module.exports = class MenteesHelper {
...
 		const totalSession = await sessionAttendeesQueries.countEnrolledSessions(id)
 
+		const userRoleIds = roles.map((role) => role.id)
+		const filter = { role_id: userRoleIds }
+		const permissionAndModules = await rolePermissionMappingQueries.find(filter)
+		const permissionsByModule = {}
+
+		permissionAndModules.forEach((rolePermission) => {
+			const module = rolePermission.dataValues.module
+			const actions = rolePermission.dataValues.actions
```
Comment (rakeshSgr) [inline src/services/mentees.js:69]: @sumanvpacewisdom change to request_type

File: src/services/mentors.js [hunk 46]
```diff
@@ -577,13 +578,35 @@ module.exports = class MentorsHelper {
...
 			const totalSession = await sessionAttendeesQueries.countEnrolledSessions(id)
 
+			const userRoleIds = roles.map((role) => role.id)
+			const filter = { role_id: userRoleIds }
+			const permissionAndModules = await rolePermissionMappingQueries.find(filter)
+			const permissionsByModule = {}
+
+			permissionAndModules.forEach((rolePermission) => {
+				const module = rolePermission.dataValues.module
+				const actions = rolePermission.dataValues.actions
```
Comment (rakeshSgr) [inline src/services/mentors.js:588]: @sumanvpacewisdom please do the request_type change, where ever you are referring to the actions

File: src/services/permissions.js [hunk 47]
```diff
@@ -25,13 +25,28 @@ module.exports = class PermissionsHelper {
...
+					return constraintError.path
+				})
+				const isCodeUnique = uniqueFields.includes('code')
+				const isApiPathUnique = uniqueFields.includes('api_path')
+
+				let errorMessage = ''
+				if (!isCodeUnique) {
+					errorMessage += 'code '
+				}
+				if (!isApiPathUnique) {
```
Comment (rakeshSgr) [inline src/services/permissions.js:44]: @sumanvpacewisdom api_path is not unique

File: src/services/mentors.js [hunk 48]
```diff
@@ -501,7 +502,7 @@ module.exports = class MentorsHelper {
 	 * @param {Boolean} isAMentor 				- user mentor or not.
 	 * @returns {JSON} 							- profile details
 	 */
-	static async read(id, orgId, userId = '', isAMentor = '') {
+	static async read(roles, id, orgId, userId = '', isAMentor = '') {
```
Comment (rakeshSgr) [inline src/services/mentors.js:505]: @sumanvpacewisdom remove roles in mentors and mentee.js files
